pkg/controller/mongodb/replica_set_controller.go

```diff
@@ -541,6 +557,24 @@ func getCurrentAutomationConfig(getUpdater secret.GetUpdater, mdb mdbv1.MongoDB)
 	return currentAc, nil
 }
 
+// validateUpdate validates that the new Spec, corresponding to the existing one
+// is still valid. If there is no a previous Spec, then the function assumes this is
+// the first version of the MongoDB resource and skips.
+func (r ReplicaSetReconciler) validateUpdate(mdb mdbv1.MongoDB) error {
+	lastSuccessfulConfigurationSaved, ok := mdb.Annotations[lastSuccessfulConfiguration]
+	if !ok {
+		// First version of Spec, no need to validate
+		return nil
+	}
+
+	last, err := mdbv1.FromJSON(lastSuccessfulConfigurationSaved)
+	if err != nil {
+		r.log.Debugf("Could not Unmarshal annotation into MongoDBSpec: ", err)
+		return nil
```

Comment:
why return nil here? If we have an annotation it should be able to be correctly read

Reply:
This function is supposed to validate the old and new specs; if for any reason the old spec can't be unmarshalled, then we assume that the actual JSON contents are still valid; for instance if the JSON document was saved with a previous version of the operator. In those cases, we must at least wait for the new spec to be saved at least once, before assuming its contents are invalid.